### PR TITLE
refactor: fresh new look for receive screen

### PR DESCRIPTION
**Pull request by sandipndev**
### TL;DR

- Refactored the underlying Payment Request objects towards a fresher more readable interface
- Added Paycode (LNURL) as a way of receiving funds
- Revamped the UI for a more cleaner look

### Payment Requests

According to the Bitcoin Design guide, the domain terminology for the object holding details of how to receive funds is called "Payment Request". Thereby, this PR very clearly has two different objects:

1. PaymentRequestCreationData
1. PaymentRequest

 Using the Creation Data, you can create a Payment Request (which is the layer that calls the GraphQL mutations). The benefit of segregation of these layers is that they can be separately tested.

### Interface

Here's how the three Invoice types (Lightning, Onchain and Paycode) look like in the app with this PR:

<table>
  <thead>
    <tr>
      <td>Lightning</td>
      <td>PayCode</td>
      <td>Onchain</td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td></td>
      <td></td>
      <td></td>
    </tr>
  </tbody>
</table>

The UI would open with a Lightning Invoice by default with the user's default wallet preselected as the receiving account. Here are the states the view could also be in depending on the situation:

<table>
  <thead>
    <tr>
      <td>Lightning Invoice Loadin</td>
      <td>Lightning Invoice Expired</td>
      <td>Invoice Paid</td>
      <td>Username unset - no Paycode</td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
    </tr>
  </tbody>
</table>

### Tasks
- [x] Delete two separate screens for usd and btc receive
- [x] Payment Requests
- [x] Expires In
- [x] Paycode
- [x] User Interface
- [x] QR View and states 
- [x] Onchain Receive Fees
- [x] Storybooks for new components
- [x] Unit Tests for Payment Request objects
- [x] End to End Test for Receive Screen
- [x] Fix Bug: Username setting doesn't trigger paycode update from page

Supersedes #2442
Fixes https://github.com/GaloyMoney/galoy-mobile/issues/2463

### CODE REVIEW

**Comment by nicolasburtey** [comment]
> ### TL;DR
> * Refactored the underlying Payment Request objects towards a fresher more readable interface
> * Added Paycode (LNURL) as a way of receiving funds
> * Revamped the UI for a more cleaner look
> 
> ### Payment Requests
> According to the Bitcoin Design guide, the domain terminology for the object holding details of how to receive funds is called "Payment Request". Thereby, this PR very clearly has two different objects:
> 
> 1. PaymentRequestCreationData
> 2. PaymentRequest
> 
> Using the Creation Data, you can create a Payment Request (which is the layer that calls the GraphQL mutations). The benefit of segregation of these layers is that they can be separately tested.
> 
> ### Interface
> Here's how the three Invoice types (Lightning, Onchain and Paycode) look like in the app with this PR:
> 
> Lightning	PayCode	Onchain
> 		
> The UI would open with a Lightning Invoice by default with the user's default wallet preselected as the receiving account. Here are the states the view could also be in depending on the situation:
> 
> Lightning Invoice Loadin	Lightning Invoice Expired	Invoice Paid	Username unset - no Paycode
> 			
> ### Tasks
> * [x]  Delete two separate screens for usd and btc receive
> * [x]  Payment Requests
> * [x]  Expires In
> * [x]  Paycode
> * [x]  User Interface
> * [x]  QR View and states
> * [x]  Onchain Receive Fees
> * [x]  Storybooks for new components
> * [x]  Unit Tests for Payment Request objects
> * [x]  End to End Test for Receive Screen
> * [x]  Fix Bug: Username setting doesn't trigger paycode update from page
> 
> Supersedes #2442 Fixes #2463

some feedback looking at the screenshots:

I don't like the fact the tap to set amount / memo height is higher than the bitcoin/stablesats and lightning/paycode/onchain are

File: e2e/05-payments-receive-flow.e2e.spec.ts [hunk 1]
```diff
@@ -210,21 +211,17 @@ describe("Receive USD Payment Flow", () => {
...
 
-  it("Wait for Green Check for USD Payment", async () => {
+  it("Wait for Green check for BTC Payment", async () => {
     const successCheck = await $(selector("Success Icon", "Other"))
-    await successCheck.waitForDisplayed({ timeout })
-  })
-
-  it("Go back to main screen", async () => {
-    await clickBackButton()
-    await waitTillOnHomeScreen()
```
Comment (nicolasburtey) [inline e2e/05-payments-receive-flow.e2e.spec.ts:228]: I would keep this test. it's useful when watching the video

File: e2e/05-payments-receive-flow.e2e.spec.ts [hunk 2]
```diff
@@ -27,7 +26,9 @@ describe("Receive BTC Amount Payment Flow", () => {
   })
 
   it("Click Request Specific Amount", async () => {
```
Comment (nicolasburtey) [inline e2e/05-payments-receive-flow.e2e.spec.ts:28]: we should certainly also tap on paycode and onchain and verify the behavior (ie: a QR code is showing up)

**Comment by UncleSamtoshi** [comment]
I think it makes more sense to have `Lightning | Paycode | Onchain` instead of `Invoice | Paycode | Onchain`. We also have `On-chain` in the button and `Onchain` in the title.

File: app/components/note-input/note-input.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    theme: { colors },
+  } = useTheme()
+
+  return (
+    <View style={[styles.fieldBackground, style]}>
+      <View style={styles.noteContainer}>
+        <TextInput
+          {...testProps("add-note")}
+          style={styles.noteInput}
+          placeholder={"Add Note"}
```
Comment (UncleSamtoshi) [inline app/components/note-input/note-input.tsx:37]: "Add note" should be translated

**Comment by UncleSamtoshi** [comment]
Going list some small follow up items:
- "Tap to set amount" has smaller font than "Add note" and the payment request types
- Why is the icon for set amount orange but the icon for "Add Note" grey?
- "Add Note" should be "Add note" to stay consistent in capitalization patterns with "Tap to set amount"

**Comment by UncleSamtoshi** [comment]
To me this looks like bitcoin is selected (even though it is actually Stablesats selected) because your eyes naturally draw to the brighter button. Any thoughts @agbegin @nicolasburtey on color of the active button? 
image

File: app/screens/receive-bitcoin-screen/receive-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+                      {LL.ReceiveScreen.shareInvoice()}
+                    </Text>
+                  </TouchableOpacity>
+                </View>
+              </>
+            )}
+        </View>
+
+        <View style={styles.extraDetails}>
+          <Text>{request.extraDetails}</Text>
```
Comment (UncleSamtoshi) [inline app/screens/receive-bitcoin-screen/receive-screen.tsx:176]: When the timer starts running its width changes due to the changing numbers. Since the extra details are centered it causes them to wiggle around.

**Comment by UncleSamtoshi** [comment]
The note icon needs some padding 
image

**Comment by UncleSamtoshi** [comment]
The text gets broken when font size is enlarged 
image

File: app/screens/receive-bitcoin-screen/payment/payment-request.ts [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+            }
+          : undefined,
+        applicationErrors: data?.onChainAddressCurrent?.errors,
+        gqlErrors: errors,
+      }
+
+      // Lightning without Amount (or zero-amount)
+    } else if (
+      pr.type === Invoice.Lightning &&
+      (pr.settlementAmount === undefined || pr.settlementAmount.amount === 0)
```
Comment (UncleSamtoshi) [inline app/screens/receive-bitcoin-screen/payment/payment-request.ts:76]: We have a `isNonZeroMoneyAmount` helper function.

File: app/screens/receive-bitcoin-screen/payment/payment-request.ts [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+              getFullUriFn,
+            }
+          : undefined,
+        applicationErrors: data?.lnUsdInvoiceCreate?.errors,
+        gqlErrors: errors,
+      }
+
+      // Paycode
+    } else if (pr.type === Invoice.PayCode && pr.username) {
+      const lnurl = await new Promise((resolve) => {
```
Comment (UncleSamtoshi) [inline app/screens/receive-bitcoin-screen/payment/payment-request.ts:204]: Why the promise here?

File: app/screens/receive-bitcoin-screen/receive-screen.tsx [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+    alignItems: "center",
+    width: 150,
+    height: 30,
+    margin: 5,
+  },
+  qrView: {
+    marginBottom: 10,
+  },
+  receivingWalletPicker: {
+    marginBottom: 10,
```
Comment (UncleSamtoshi) [inline app/screens/receive-bitcoin-screen/receive-screen.tsx:269]: For future reference, instead of spacing everything with margin, you can set the `rowGap` property on the top level container.

File: e2e/05-payments-receive-flow.e2e.spec.ts [hunk 8]
```diff
@@ -219,18 +229,119 @@ describe("Receive USD Payment Flow", () => {
...
+      // work on physical device for security reasons
+      const shareButton = await $(selector("Share Invoice", "StaticText"))
+      await shareButton.waitForDisplayed({ timeout })
+      await shareButton.click()
+      const invoiceSharedScreen = await $('//*[contains(@name,"bitcoin:tb1")]')
+      await invoiceSharedScreen.waitForDisplayed({
+        timeout: 8000,
+      })
+      invoice = await invoiceSharedScreen.getAttribute("name")
+      const closeShareButton = await $(selector("Close", "Button"))
```
Comment (UncleSamtoshi) [inline e2e/05-payments-receive-flow.e2e.spec.ts:299]: There should be helper functions for a lot of the actions you are trying to take such as clicking a button. You can find these in the `e2e/utils/controls.ts` and `e2e/utils/use-cases.ts`

**Review by UncleSamtoshi — APPROVED** [review]
Excited for this to be rolled out. Great job!

**Comment by nicolasburtey** [comment]
Sounds good to me to merge and get feedback on the live internal app

File: app/screens/receive-bitcoin-screen/receive-screen.tsx (hunk 4 above)
Comment (sandipndev) [reply app/screens/receive-bitcoin-screen/receive-screen.tsx:176]: How should we go about this problem?
